src/recipe/recipe.controller.ts

```diff
@@ -64,19 +67,46 @@ export class RecipeController {
   })
   @Get(':id')
   async fetchRecipe(
-    @Param('id', ParseIntPipe) id: number,
+    @UserId() userId: number,
+    @Param('id', ParseIntPipe) recipeId: number,
   ): Promise<FetchRecipeResponse> {
-    const fetchedRecipe = await this.recipeService.fetchRecipe(id);
+    const fetchedRecipe = await this.recipeService.fetchRecipe(
+      recipeId,
+      userId,
+    );
+
     return FetchRecipeResponse.from(fetchedRecipe);
   }
 
+  @HttpCode(200)
+  @UseGuards(JwtAuthGuard)
+  @ApiOperation({ summary: 'Get all users recipes' })
+  @ApiBearerAuth()
+  @ApiUnauthorizedResponse({ description: 'User is not authenticated' })
+  @ApiParam({
+    name: 'email',
+    description: 'Email of a user',
+  })
+  @Get('/user/:id')
+  async fetchRecipesByAuthorId(
+    @UserId() userId: number,
+    @Param('id') authorId: number,
+  ): Promise<FetchRecipesResponse> {
+    const fetchedUsersRecipes = await this.recipeService.fetchRecipesByAuthorId(
+      authorId,
+      userId,
+    );
+
+    return FetchRecipesResponse.from(fetchedUsersRecipes);
+  }
+
```

Comment:
hmm, I think I wouldn't create a separate endpoint for this. Instead in the existing `GET /recipes` endpoint I would add support for filters in query params, e.g. `GET /recipes?authorId=123456`